### PR TITLE
 feat: add hcloud_iso_info module

### PR DESCRIPTION
**Pull request by jooola**
##### SUMMARY

Resurrecting #145 

Created module hcloud_iso_info.

##### ISSUE TYPE

- Feature Pull Request

##### COMPONENT NAME

`hcloud_iso_info`

### CODE REVIEW

File: tests/integration/targets/hcloud_iso_info/meta/main.yml [hunk 1]
```diff
@@ -0,0 +1,3 @@
+collections:
+  - community.general.ipfilter
```
Comment (apricote) [inline tests/integration/targets/hcloud_iso_info/meta/main.yml:2]: What is the ipfilter used for?

File: plugins/modules/hcloud_iso_info.py [hunk 2]
```diff
@@ -0,0 +1,167 @@
...
+    - Lukas Kaemmerling (@LKaemmerling)
+
+options:
+    id:
+        description:
+            - The ID of the ISO image you want to get.
+        type: int
+    name:
+        description:
+            - The description of the ISO you want to get.
```
Comment (apricote) [inline plugins/modules/hcloud_iso_info.py:28]:
```suggestion
            - The name of the ISO you want to get.
```

File: plugins/modules/hcloud_iso_info.py [hunk 3]
```diff
@@ -0,0 +1,167 @@
...
+        description:
+            description: Description of the ISO
+            returned: always
+            type: str
+            sample: Debian 12.0 (amd64/netinstall)
+        architecture:
+            description: >
+                Type of cpu architecture this ISO is compatible with.
+                None indicates no restriction on the architecture (wildcard).
+            returned: always
```
Comment (apricote) [inline plugins/modules/hcloud_iso_info.py:80]: This field is actually not required and can be `null` for `private` isos

File: tests/integration/targets/hcloud_iso_info/aliases [hunk 4]
```diff
@@ -0,0 +1,2 @@
+cloud/hcloud
+shippable/hcloud/group2
```
Comment (apricote) [inline tests/integration/targets/hcloud_iso_info/aliases:2]: I suggest `group3`, its currently the emptiest -> does not cause longer CI runs.

```suggestion
shippable/hcloud/group3
```

File: tests/integration/targets/hcloud_iso_info/meta/main.yml (hunk 1 above)
Comment (jooola) [reply tests/integration/targets/hcloud_iso_info/meta/main.yml:2]: This is a good question, I copied this from another test case.
- The collection name seems invalid anyway,
- The name/function is not used anywhere.

I'll try to find out why we have this.

File: plugins/modules/hcloud_iso_info.py (hunk 3 above)
Comment (jooola) [reply plugins/modules/hcloud_iso_info.py]: I hope the wording is correct with the new value `when relevant`.

**Review by apricote — APPROVED** [review]
Thanks for picking this up!

**Pull request merged**